fifo_test.go

```diff
@@ -42,15 +42,16 @@ func TestFifoCancel(t *testing.T) {
 		leakCheckWg = nil
 	}()
 
-	_, err = OpenFifo(context.Background(), filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_NONBLOCK, 0600)
+	f, err := OpenFifo(context.Background(), filepath.Join(tmpdir, "f0"), syscall.O_RDONLY|syscall.O_NONBLOCK, 0600)
+	assert.Exactly(t, nil, f)
```

Comment:
Annoyingly, `assert.Nil` considers non-nil interface values which contain nil concrete values to be nil values despite those values not being `== nil`.